### ISSUE TITLE
Any plan for supporting libusb_init_context()?

### ISSUE THREAD

**Issue by zephyr-fun** [a1ien/rusb#207]
Thanks for your brilliant work! It really paves the way between rust and USB devices. 

TL:DR: is there any plan to support calling libusb_set_option() before libusb_init(), which is super useful in access-limited situations?

For now, the crate works fine in most situations, however, when it comes to circumstances where **access to USB devices is limited**, there seems to be no way to call **libusb_set_option()** with opt equals to **LIBUSB_OPTION_NO_DEVICE_DISCOVERY** before **libusb_init()**, which helps the caller to skip scanning devices when calling **libsub_init()**. 
Also, the libusb_init() API has already been marked as **deprecated** , it would be a better choice to turn to libusb_init_context().

Looking forward to your replies!

You may find the following links helpful:
https://libusb.sourceforge.io/api-1.0/group__libusb__lib.html#ga7deaef521cfb1a5b3f8d6c01be11a795
https://libusb.sourceforge.io/api-1.0/group__libusb__lib.html#ga07d4ec54cf575d672ba94c72b3c0de7c

**Comment by a1ien** [comment]
Sorry. For long delay with answer. Yes I can try to add this possibility. But this Maybe A little tricky.

**Comment by dgramop** [comment]
See: #209 :)

**Comment by a1ien** [comment]
Actually for android you can use old approache
```
libusb1_sys::libusb_set_option(
        std::ptr::null_mut(),
        LIBUSB_OPTION_NO_DEVICE_DISCOVERY
    )
```
or `rusb::disable_device_discovery`

**Comment by dgramop** [comment]
the old libusb_set_option way would trigger segmentation faults on Android, but libusb_init_context with the same option would not. I think it has to do with how set_option is implemented and some faulty assumption about memory allocation or struct padding that doesn't hold up in my environment

**Comment by a1ien** [comment]
The old `libusb_set_option` should still work. You must call it before any operation.

**Comment by dgramop** [comment]
it seems i was doing this incorrectly. i had mis-re-typed the original sample code in the android directory when writing the C++ sanity-check proof-of-concept code on android (which also segfaulted till i switched to the new API). that behavior is a little unintuitive, i'm glad they deprecated that API haha. closer reading may have saved me.
